File: base/animal2/.cache/adapter.java
public class Adapter {
    private Animal animal;
    public Adapter(String especie, String barulho) {
        animal = new Animal(especie, barulho);
    }
    
    public String noise() {
        return animal.fazerBarulho();
    }

    public void grow(int nivel) {
        animal.envelhecer(nivel);
    }

    public String toString() {
        return animal.toString();
    }
}

File: base/animal2/.cache/cheat/ts/adapter.ts
import {Animal} from "./animal";

class Adapter {
    animal: Animal;

    constructor(especie: string, barulho: string) {
        this.animal = new Animal(especie, barulho);
    }

    grow(nivel: number): void {
        this.animal.envelhecer(nivel);
    }

    noise(): string {
        return this.animal.fazerBarulho();
    }

    show(): string {
        return this.animal.toString();
    }
}

export {Adapter};
